# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/jimmy/nullmailer/daemon.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./src/send.cc.orig	2003-01-03 22:58:31.000000000 +0200
+++ ./src/send.cc	2003-06-15 10:06:30.000000000 +0300
@@ -319,10 +319,19 @@
   signal(SIGHUP, SIG_IGN);
   load_config();
   load_files();
-  for(;;) {
+  if(pausetime == 0)
     send_all();
-    if (pausetime == 0) break;
-    do_select();
+  else {
+    switch(fork()) {
+      case 0:
+	for(;;) {
+ 	  send_all();
+ 	  do_select();
+	}
+      case -1:
+	fout << "Could not fork main send message loop." << endl;
+	return 1;
+    }
   }
   return 0;
 }
